nixos/modules/services/logging/journalwatch.nix

```diff
@@ -43,6 +56,15 @@ in {
         '';
       };
 
+      package = mkOption {
```

Comment:
I only see one version of this package in nixpkgs. Is this option necessary?

Reply:
IIRC, the way this should be done is via [overlays or packageOverrides](https://github.com/NixOS/nixpkgs/issues/55366#issuecomment-461780866) (the latter being [discouraged](https://github.com/NixOS/nixpkgs/pull/43560))

Reply:
I see; a working way to override the single pkgs.journalwatch with another version would indeed be enough and make this option unnecessary. Already did some research in that direction in the past, but never got it to work.

Could someone please give me a hand and tell me what I'm doing wrong here? All examples other than #55366 that I found only did add new packages or a full package set, but did not *replace* an existing package.

```nix
let
  mypkgsSource = builtins.fetchGit {
    # Descriptive name to make the store path easier to identify
    name = "florianjacob-nixpkgs";
    url = https://github.com/florianjacob/nixpkgs/;
    ref = "journalwatch";
    rev = "d5e303c9df8da2ae4aa42a7ccd28fbae4d24e30d";
  };
  mypkgs = import mypkgsSource {
    # does not help either
    # config = removeAttrs config.nixpkgs.config [ "packageOverrides" ];
  };
in
{
  # does not make a difference
  # nixpkgs.config.packageOverrides = pkgs: {
  #   # python3.pkgs.journalwatch = mypkgs.python3.pkgs.journalwatch;
  # };
  nixpkgs.overlays = [ (self: super: {
    python3.pkgs.journalwatch = mypkgs.python3.pkgs.journalwatch;
    # same error
    # python3Packages.journalwatch = mypkgs.python3Packages.journalwatch;
  }) ];

  # works fine:
  # services.journalwatch = {
  #   enable = true;
  #   package = mypkgs.python3Packages.journalwatch;
  # };
};
```

leads to

```
error: attribute 'buildPythonApplication' missing, at /nix/store/ai3gypqdhy8as9pik587cf7yx35546m2-nixos-19.03.172138.5c52b25283a/nixos/pkgs/development/tools/build-managers/meson/default.nix:3:1
```

Reply:
@florianjacob sorry for taking a while to get back to you on this. Here is what I used:
```
  nixpkgsFJ = builtins.fetchGit {
    # Descriptive name to make the store path easier to identify
    name = "florianjacob-nixpkgs";
    url = https://github.com/florianjacob/nixpkgs/;
    ref = "journalwatch";
    rev = "60b9e339059e72d6ac3469509c41dacca57b2979";
  };
  pkgsFJ = import nixpkgsFJ { config = {}; };
```
...
```
  nixpkgs.overlays = [
    (self: super: {
      python3 = super.python3.override {
        packageOverrides = python-self: python-super: {
          journalwatch = pkgsFJ.python3Packages.journalwatch;
        };
      };
    })
  ];
```

Reply:
Works indeed, thanks alot! Removed the option now.